typing/typedecl.ml

```diff
@@ -411,7 +407,10 @@ let transl_declaration env sdecl (id, uid) =
         type_unboxed = unboxed_status;
         type_uid = uid;
       } in
-
+    begin match Ctype.check_decl_immediate env decl immediate with
+    | Ok () -> ()
+    | Error v -> raise(Error(sdecl.ptype_loc, Immediacy v))
+    end;
```

Comment:
It may be useful to move this check below the "check for cyclic abbreviations" check, in case a cyclic abbreviations would break the immediacy-computation logic by running out of fuel during unboxed expansion.

Reply:
Well spotted! I'll try to construct a test case that hits this horrible case.